docs/CUSTOM_BOARDS_TARGETS.md

```diff
@@ -0,0 +1,61 @@
+# mLRS Documentation: Custom Boards & Targets #
+
+([back to main page](../README.md))
+
+While mLRS offers a variety of hardware options, you may find it beneficial for your application to create a custom board to better suit your objectives.  This page describes how to use the existing codebase to enable the use of custom boards and custom targets.
+
+## Definitions
+
+- 'Board' refers to the physical hardware
+- 'Target' refers to the firmware that will run on the board
+
+## Assumptions
+
+1. You have setup the build environment for mLRS
+2. You are comfortable with C/C++ programming
+
+## Creating a target for a custom board
+
+There are a number of ways of how to create a target for a custom board, for example:
+- A simple way would be to identify an existing board which is close to your custom board and modify the entries in this boards hal file.
+- A more advanced way would be to create a new target as you find them for the existing boards. This requires more work and additional skills.
+
+The suggested procedure is simple and makes it easier to update the code base by seperating out changes:
+ 
+1. Identify a target which is close to your new board. Example: The `rx-diy-board01-f103cb` target may be it.
+    1. Note: the only condition when picking a target is that you will need to choose a target which uses the same MCU as your custom board, in this example it is assumed that you're using the STM32F103CBXX on your custom board.
+2. Copy the existing target's hal file and rename it to what you want it to be. Example: Copy `/Common/hal/rx-hal-diy-board01-f103cb.h` to `/Common/hal/rx-hal-diy-customboard-f103cb.h`.
+3. In `/Common/hal/hal.h` go to the area where the existing target's hal file is included, comment out the include, and add a line which inlcudes your hal file. Example: Find the line with `#include "rx-hal-diy-board01-f103cb.h"`, replace it with `\\#include "rx-hal-diy-board01-f103cb.h"` and insert a line `#include "rx-hal-diy-customboard-f103cb.h"`.
+4. In `/Common/hal/device_conf.h` go to the area where the existing target is defined, comment out the define of the device name, and add your own define changing the device name to what you want it to be. Example: Find the line with `#ifdef RX_DIY_BOARD01_F103CB`, replace the subsequent line with `\\#define DEVICE_NAME "DIY DualSX F103CB"` and insert a line `#define DEVICE_NAME "My Own Great Board"`
+    1. Note: the device name can be 20 characters max.
+5. Modify the target's device_conf entry and hal file to match the features of your custom board - further detailed below.
+
+## Modifying a target
+
+The codebase has two locations that defines the features and capabilites of a target:
+1. Device Conf
+    1. Defines the device name, type (Tx/Rx), Sx chipset, frequencies
+    2. Location: mLRS/mLRS/Common/hal/device_conf.h
+        
+2. Target Hal
+    1. Defines the features, UART assignment and pin mapping
+    2. Location: mLRS/mLRS/Common/hal/TARGET.h
```

Comment:
TARGET is a bit strange here, maybe my-target. Maybe add an example, like inteh above.

Reply:
Updated to 'MY-CUSTOM-TARGET.h'